espnet2/asr/partially_AR_model.py

```diff
@@ -0,0 +1,247 @@
+import logging
+import warnings
+from contextlib import contextmanager
+from itertools import groupby
+from typing import Dict, List
+
+warnings.filterwarnings("ignore", category=UserWarning)
+
+import numpy
+import torch
+from packaging.version import parse as V
+
+from espnet2.asr.ctc import CTC
+from espnet2.asr.decoder.abs_decoder import AbsDecoder
+from espnet2.text.build_tokenizer import build_tokenizer
+from espnet2.text.token_id_converter import TokenIDConverter
+from espnet.nets.beam_search import Hypothesis
+from espnet.nets.beam_search_partially_AR import PartiallyARBeamSearch
+from espnet.nets.scorer_interface import ScorerInterface
+
+if V(torch.__version__) >= V("1.6.0"):
+    from torch.cuda.amp import autocast
+else:
+    # Nothing to do if torch<1.6.0
+    @contextmanager
+    def autocast(enabled=True):
+        yield
+
+
+class PartiallyARInference(torch.nn.Module):
+    """Mask-CTC-based partially autoregressive inference"""
+
+    def __init__(
+        self,
+        ctc: CTC,
+        decoder: AbsDecoder,
+        threshold_probability: float,
+        sos: int = 4999,
+        eos: int = 4999,
+        mask_token: int = 5000,

```

Comment:
Do we need these default values? They might depend on the specific model? Can we use None instead?